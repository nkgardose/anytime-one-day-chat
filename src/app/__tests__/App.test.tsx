import { act, fireEvent, render, screen } from '@testing-library/react'
import App from '..'
import Providers from '../../providers'
import '@testing-library/jest-dom'

describe('App', () => {
  it('renders properly', () => {
    const { container } = render(
      <Providers>
        <App />
      </Providers>
    )

    expect(container).toMatchSnapshot()
  })
  it('renders the correct content', () => {
    render(
      <Providers>
        <App />
      </Providers>
    )

    expect(screen.getByText('Joyse')).toBeInTheDocument()
    expect(screen.getByText('General Channel')).toBeInTheDocument()
    expect(screen.getByText('Technology Channel')).toBeInTheDocument()
    expect(screen.getByText('LGTM Channel')).toBeInTheDocument()
  })
  it('navigates to the correct channel', async () => {
    render(
      <Providers>
        <App />
      </Providers>
    )
    await act(async () => {
      fireEvent.click(screen.getByText('General Channel'))
    })

    expect(screen.getAllByText('General Channel').length).toBe(2)
  })
})
