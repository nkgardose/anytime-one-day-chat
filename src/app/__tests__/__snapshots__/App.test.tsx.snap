// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders properly 1`] = `
<div>
  <div
    class="css-1xj4huh-Layout"
  >
    <div
      class="css-1e86s6t-Channel"
    >
      <nav
        class="css-17wuva9-Header"
      >
        <div
          class="css-1ysghxv-Users"
        >
          <select
            class="css-13amir5-Users"
          >
            <option>
              Joyse
            </option>
            <option>
              Sam
            </option>
            <option>
              Russell
            </option>
          </select>
          <i
            aria-label="Dropdown"
            class="css-1mpnkv5-Users"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M24 24H0V0h24v24z"
                fill="none"
                opacity=".87"
              />
              <path
                d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6-1.41-1.41z"
              />
            </svg>
          </i>
        </div>
      </nav>
      <ul
        class="css-1ti0mfk-ChannelList"
      >
        <li
          aria-hidden="true"
          class="css-1oyhoxx-ChannelItem"
        >
          General
           Channel
        </li>
        <li
          aria-hidden="true"
          class="css-1oyhoxx-ChannelItem"
        >
          Technology
           Channel
        </li>
        <li
          aria-hidden="true"
          class="css-1oyhoxx-ChannelItem"
        >
          LGTM
           Channel
        </li>
      </ul>
    </div>
    <div
      class="css-1cohcow-Layout"
    >
      <div>
        wala
      </div>
    </div>
  </div>
</div>
`;
